src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,267 @@
+        {
+            var F = new SyntheticBoundNodeFactory(this, ContainingType.GetNonNullSyntaxNode(), compilationState, diagnostics);
+            try
+            {
+                ImmutableArray<Symbol> printableMembers = ContainingType.GetMembers()
+                    .WhereAsArray(m => m.DeclaredAccessibility == Accessibility.Public && (m.Kind is SymbolKind.Field or SymbolKind.Property));
+
+                if (ReturnType.IsErrorType() ||
+                    printableMembers.Any(m => m.GetTypeOrReturnType().Type.IsErrorType()))
+                {
+                    F.CloseMethod(F.ThrowNull());
+                    return;
+                }
+
+                ArrayBuilder<BoundStatement>? block = printableMembers.IsEmpty ? null : ArrayBuilder<BoundStatement>.GetInstance();
+                BoundParameter builder = F.Parameter(this.Parameters[0]);
+                if (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType())
+                {
+                    if (printableMembers.IsEmpty)
+                    {
+                        // return false;
+                        F.CloseMethod(F.Return(F.Literal(false)));
+                        return;
+                    }
+                }
+                else
+                {
+                    MethodSymbol? printMethod = FindValidPrintMembersMethod(ContainingType.BaseTypeNoUseSiteDiagnostics, DeclaringCompilation);
+                    if (printMethod is null)
+                    {
+                        F.CloseMethod(F.ThrowNull()); // an error was reported in base checks already
+                        return;
+                    }
+
+                    var basePrintCall = F.Call(receiver: F.Base(ContainingType.BaseTypeNoUseSiteDiagnostics), printMethod, builder);
+                    if (printableMembers.IsEmpty)
+                    {
+                        // return base.print(builder);
+                        F.CloseMethod(F.Return(basePrintCall));
+                        return;
+                    }
+                    else
+                    {
+                        // if (base.print(builder))
+                        //     builder.Append(", ")
+                        block!.Add(F.If(basePrintCall, makeAppendString(F, builder, ", ")));
+                    }
+                }
+
+                Debug.Assert(!printableMembers.IsEmpty && block is object);
+
+                for (var i = 0; i < printableMembers.Length; i++)
+                {
+                    // builder.Append(<name>);
+                    // builder.Append(" = ");
+                    // builder.Append((object)<value>);
+                    // builder.Append(", "); // except for last member
+
+                    var member = printableMembers[i];
+                    block.Add(makeAppendString(F, builder, member.Name));
+                    block.Add(makeAppendString(F, builder, " = "));
```

Comment:
The string "Name = " could be calculated at compile time: `makeAppendString(F, builder, member.Name + " = ")`